spec/acceptance/autoscaling_spec.rb

```diff
@@ -32,39 +32,64 @@ def find_alarm(name)
+      @sg_delete = 'sg_delete.pp.tmpl'
+      # launch asg and related resources
+      @asg_config = {
+        :ensure               => 'present',
+        :region               => @default_region,
+        :sg_name              => "#{name}-sg",
+        :lc_name              => "#{name}-lc",
+        :sg_setting           => "#{name}-sg",
+        :asg_name             => "#{name}-asg",
+        :min_size             => 2,
+        :max_size             => 6,
+        :lc_setting           => "#{name}-lc",
+        :availability_zones   => ["#{@default_region}a", "#{@default_region}b"],
+        :policy_name          => "#{name}-policy",
+        :second_policy_name   => "#{name}-second_policy",
+        :asg_setting          => "#{name}-asg",
+        :scaling_adjustment   => 30,
+        :adjustment_type      => 'PercentChangeInCapacity',
+        :alarm_name           => "#{name}-cw_alarm",
+        :metric               => 'CPUUtilization',
+        :namespace            => 'AWS/EC2',
+        :statistic            => 'Average',
+        :period               => 120,
+        :threshold            => 70,
+        :comparison_operator  => 'GreaterThanOrEqualToThreshold',
+        :asg_setting          => "#{name}-asg",
+        :evaluation_periods   => 2,
+        :alarm_actions        => "#{name}-policy",
       }
-      @config[:optional] = {:key_name => ENV['AWS_KEY_PAIR']} if ENV['AWS_KEY_PAIR']
-
-      @template = 'autoscaling.pp.tmpl'
-      PuppetManifest.new(@template, @config).apply
+      r = PuppetManifest.new(@asg_template, @asg_config).apply
+      expect(r[:output].any?{ |o| o.include?('Error:')}).to eq(false)
+      # launch duplicate resources
+      @duplicate_asg_config = {
+        :region       => @default_region,
+        :sg_name      => "#{name}-sg2",
+        :lc_name      => "#{name}-lc2",
+      }
+      r2 = PuppetManifest.new(@duplicate_asg_template, @duplicate_asg_config).apply
+      expect(r2[:output].any?{ |o| o.include?('Error:')}).to eq(false)
     end
 
     after(:all) do
-      new_config = @config.update({:ensure => 'absent'})
-      template = 'autoscaling_delete.pp.tmpl'
-      PuppetManifest.new(template, new_config).apply
-      # wait for all instances under ASG to be terminated
+      #audit this entire teardown
+      @asg_config[:ensure] = 'absent'
+      @duplicate_asg_config[:ensure] = 'absent'
+      duplicate_delete = 'duplicate_asg_delete.pp.tmpl'
+      r = PuppetManifest.new(@asg_template_delete, @asg_config).apply
+      # assert that none of the results contain 'Error:'
+      expect(r[:output].any?{ |o| o.include?('Error:')}).to eq(false)
       response = @aws.autoscaling_client.describe_auto_scaling_groups(
-        auto_scaling_group_names: ["#{@name}-asg"],
+        auto_scaling_group_names: [@asg_config[:asg_name]],
       )
       id = Array.new
       response.to_h[:auto_scaling_groups].first[:instances].each do |x|
```

Comment:
While running this I noticed that this line fails with an exception, but doesn't cause the tests to fail as it's in the after block. It was added in https://github.com/puppetlabs/puppetlabs-aws/commit/e7ae82830ae108a15eb5afc5dbe8e235e065f334

Specifically, to_h fails with `TypeError: wrong element type Aws::PageableResponse (expected array)`. The fix is to access the data directly like so:

```ruby
response.data[:auto_scaling_groups].first[:instances].each do |x|
```